iscsi-scst/usr/event.c

```diff
@@ -1094,6 +1114,53 @@ static int handle_e_set_attr_value(int fd, const struct iscsi_kern_event *event)
 			res = -EINVAL;
 			goto out_free;
 		}
+	} else if (strcasecmp(ISCSI_INTERNAL_PORTAL_ATTR_NAME, pp) == 0) {
+		if (target != NULL) {
+			log_error("Not NULL target %s for global attribute %s",
+				target->name, pp);
+			res = -EINVAL;
+			goto out_free;
+		}
+		p = config_strip_string(p);
+		if (!p || *p == '\0') {
+			if (internal_portal)
```

Comment:
Unnecessary check before free(). You can pass NULL to free(), so this check is not needed.